*** Starting uWSGI 2.0.29 (64bit) on [Thu May 15 10:48:22 2025] ***
compiled with version: 12.2.0 on 29 April 2025 11:17:30
os: Linux-6.1.0-27-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.115-1 (2024-11-01)
nodename: jsbdebian
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 12
current working directory: /home/jsbien/git/glyph-miner
detected binary path: /home/jsbien/git/glyph-miner/uwsgi-env/bin/uwsgi
chdir() to /home/jsbien/git/glyph-miner
your processes number limit is 62262
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:9091 fd 3
Python version: 3.11.2 (main, Nov 30 2024, 21:22:50) [GCC 12.2.0]
Python main interpreter initialized at 0x7f43a0760018
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 166752 bytes (162 KB) for 2 cores
*** Operational MODE: threaded ***
added /home/jsbien/git/glyph-miner/ to pythonpath.
✅ server.webapp.__init__.py loaded
>>> LOADED: /home/jsbien/git/glyph-miner/server/server.py ver. 0.05 <<<
[DEBUG] web.application: True
>>> collections_handler has methods: ['GET', 'POST', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__']
[DEBUG] collections = <class 'server.server.collections_handler'> (type: <class 'type'>)
>>> collections_handler has POST: True
>>> YES: collections_handler.POST exists
[DEBUG] application = <function application.wsgifunc.<locals>.wsgi at 0x7f439cefa2a0>
[DEBUG] application.__module__ = server.webapp.application
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x7f43a0760018 pid: 1706818 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 1706818)
spawned uWSGI worker 1 (pid: 1706834, cores: 2)
[DEBUG] resolve_route() called with path: /api/ping
[DEBUG] resolve_route() called with path: /api/ping
[DEBUG] 🐒 f = PingHandler (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.PingHandler'> (type: <class 'type'>)
📡 header() called with: Content-Type=text/plain, ctx.headers type: <class 'list'>
[pid: 1706834|app: 0|req: 1/1] 127.0.0.1 () {32 vars in 368 bytes} [Thu May 15 10:48:22 2025] GET /api/ping => generated 4 bytes in 2 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 0)
[DEBUG] resolve_route() called with path: /api/ping
[DEBUG] resolve_route() called with path: /api/ping
[DEBUG] 🐒 f = PingHandler (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.PingHandler'> (type: <class 'type'>)
📡 header() called with: Content-Type=text/plain, ctx.headers type: <class 'list'>
[pid: 1706834|app: 0|req: 2/2] 127.0.0.1 () {32 vars in 368 bytes} [Thu May 15 10:48:22 2025] GET /api/ping => generated 4 bytes in 0 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 1)
[DEBUG] resolve_route() called with path: /api/images
[DEBUG] resolve_route() called with path: /api/images
[DEBUG] 🐒 f = images (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.images'> (type: <class 'type'>)
📡 header() called with: Access-Control-Allow-Origin=*, ctx.headers type: <class 'list'>
>>> ENTERED images.GET()
'🔍 Attempting to fetch images with collection count'
>>> QUERY: SELECT i.*, COUNT(ci.collection_id) AS collection_count FROM images i LEFT OUTER JOIN collections_images ci ON i.id = ci.image_id GROUP BY i.id
>>> VARS: None
[DEBUG] resolve_route() called with path: /api/collections
>>> Reconnecting to MySQL
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] 🐒 f = collections (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.collections_handler'> (type: <class 'type'>)
>>> ENTERED collection_handler GET <<<
>>> QUERY: SELECT * FROM collections
>>> VARS: None
>>> Reconnecting to MySQL
>>> CURSOR CLASS: <class 'server.database.connection.PatchedDictCursor'>
>>> CURSOR CLASS: <class 'server.database.connection.PatchedDictCursor'>
>>> QUERY SUCCESS
>>> FETCHED: []
📡 header() called with: Content-Type=application/json, ctx.headers type: <class 'list'>
[pid: 1706834|app: 0|req: 4/3] ::1 () {52 vars in 864 bytes} [Thu May 15 10:48:23 2025] GET /api/collections => generated 2 bytes in 1 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 1)
>>> QUERY SUCCESS
'✅ Images fetched successfully'
[pid: 1706834|app: 0|req: 4/4] ::1 () {52 vars in 854 bytes} [Thu May 15 10:48:23 2025] GET /api/images => generated 2 bytes in 2 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 0)
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] 🐒 f = collections (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.collections_handler'> (type: <class 'type'>)
>>> ENTERED collection_handler GET <<<
>>> QUERY: SELECT * FROM collections
>>> VARS: None
>>> Reconnecting to MySQL
>>> CURSOR CLASS: <class 'server.database.connection.PatchedDictCursor'>
>>> QUERY SUCCESS
>>> FETCHED: []
📡 header() called with: Content-Type=application/json, ctx.headers type: <class 'list'>
[pid: 1706834|app: 0|req: 5/5] ::1 () {52 vars in 864 bytes} [Thu May 15 10:48:23 2025] GET /api/collections => generated 2 bytes in 0 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 1)
[DEBUG] resolve_route() called with path: /api/debug/clear
[DEBUG] resolve_route() called with path: /api/debug/clear
[DEBUG] 🐒 f = DebugClearHandler (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.DebugClearHandler'> (type: <class 'type'>)
>>> CLEARING collections table <<<
[pid: 1706834|app: 0|req: 6/6] 127.0.0.1 () {32 vars in 383 bytes} [Thu May 15 10:48:24 2025] POST /api/debug/clear => generated 27 bytes in 0 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 0)
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] 🐒 f = collections (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.collections_handler'> (type: <class 'type'>)
>>> ENTERED POST <<<
[ERROR] POST failed: 'env'
📦 HTTPError.init: headers = [('Content-Type', 'text/html')] (<class 'list'>)
DEBUG FULL HEADERS LIST: [('Content-Type', 'text/html')]
DEBUG FULL HEADERS TYPES: [(<class 'str'>, <class 'str'>)]
📡 header() called with: Content-Type=text/html, ctx.headers type: <class 'list'>
Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 370, in data
    return ctx.data
           ^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'data'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/server.py", line 133, in POST
    raw_data = web.data()
               ^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 373, in data
    length = int(ctx.env.get('CONTENT_LENGTH', 0))
                 ^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'env'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 78, in handle_with_processors
    return self._delegate(handler_key, self.fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 120, in _delegate
    return self._delegate(cls, fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 105, in _delegate
    return handle_class(f)
           ^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 98, in handle_class
    return tocall(*args)
           ^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/server.py", line 159, in POST
    raise web.internalerror()
server.webapp.webapi._InternalError

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 370, in data
    return ctx.data
           ^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'data'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/server.py", line 133, in POST
    raw_data = web.data()
               ^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 373, in data
    length = int(ctx.env.get('CONTENT_LENGTH', 0))
                 ^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'env'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 48, in wsgi
    result = self.handle_with_processors()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 78, in handle_with_processors
    return self._delegate(handler_key, self.fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 120, in _delegate
    return self._delegate(cls, fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 105, in _delegate
    return handle_class(f)
           ^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 98, in handle_class
    return tocall(*args)
           ^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/server.py", line 159, in POST
    raise web.internalerror()
server.webapp.webapi._InternalError

[pid: 1706834|app: 0|req: 7/7] ::1 () {40 vars in 538 bytes} [Thu May 15 10:48:24 2025] POST /api/collections => generated 21 bytes in 2 msecs (HTTP/1.1 500) 1 headers in 64 bytes (1 switches on core 1)
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] 🐒 f = collections (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.collections_handler'> (type: <class 'type'>)
>>> ENTERED collection_handler GET <<<
>>> QUERY: SELECT * FROM collections
>>> VARS: None
>>> Reconnecting to MySQL
>>> CURSOR CLASS: <class 'server.database.connection.PatchedDictCursor'>
>>> QUERY SUCCESS
>>> FETCHED: []
📡 header() called with: Content-Type=application/json, ctx.headers type: <class 'list'>
[pid: 1706834|app: 0|req: 8/8] ::1 () {36 vars in 457 bytes} [Thu May 15 10:48:25 2025] GET /api/collections => generated 2 bytes in 1 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 0)
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] 🐒 f = collections (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.collections_handler'> (type: <class 'type'>)
>>> ENTERED collection_handler GET <<<
>>> QUERY: SELECT * FROM collections
>>> VARS: None
>>> Reconnecting to MySQL
>>> CURSOR CLASS: <class 'server.database.connection.PatchedDictCursor'>
>>> QUERY SUCCESS
>>> FETCHED: []
📡 header() called with: Content-Type=application/json, ctx.headers type: <class 'list'>
[pid: 1706834|app: 0|req: 9/9] ::1 () {36 vars in 457 bytes} [Thu May 15 10:48:25 2025] GET /api/collections => generated 2 bytes in 1 msecs (HTTP/1.1 200) 1 headers in 44 bytes (1 switches on core 1)
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] resolve_route() called with path: /api/collections
[DEBUG] 🐒 f = collections (type: <class 'str'>)
[DEBUG] 🐒 f = <class 'server.server.collections_handler'> (type: <class 'type'>)
>>> ENTERED POST <<<
[ERROR] POST failed: 'env'
📦 HTTPError.init: headers = [('Content-Type', 'text/html')] (<class 'list'>)
DEBUG FULL HEADERS LIST: [('Content-Type', 'text/html')]
DEBUG FULL HEADERS TYPES: [(<class 'str'>, <class 'str'>)]
📡 header() called with: Content-Type=text/html, ctx.headers type: <class 'list'>
Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 370, in data
    return ctx.data
           ^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'data'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/server.py", line 133, in POST
    raw_data = web.data()
               ^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 373, in data
    length = int(ctx.env.get('CONTENT_LENGTH', 0))
                 ^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'env'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 78, in handle_with_processors
    return self._delegate(handler_key, self.fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 120, in _delegate
    return self._delegate(cls, fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 105, in _delegate
    return handle_class(f)
           ^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 98, in handle_class
    return tocall(*args)
           ^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/server.py", line 159, in POST
    raise web.internalerror()
server.webapp.webapi._InternalError

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 370, in data
    return ctx.data
           ^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'data'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/server.py", line 133, in POST
    raw_data = web.data()
               ^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/webapi.py", line 373, in data
    length = int(ctx.env.get('CONTENT_LENGTH', 0))
                 ^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/utils.py", line 671, in __getattr__
    raise KeyError(key)
KeyError: 'env'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 48, in wsgi
    result = self.handle_with_processors()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 78, in handle_with_processors
    return self._delegate(handler_key, self.fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 120, in _delegate
    return self._delegate(cls, fvars, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 105, in _delegate
    return handle_class(f)
           ^^^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/webapp/application.py", line 98, in handle_class
    return tocall(*args)
           ^^^^^^^^^^^^^
  File "/home/jsbien/git/glyph-miner/server/server.py", line 159, in POST
    raise web.internalerror()
server.webapp.webapi._InternalError

[pid: 1706834|app: 0|req: 10/10] ::1 () {40 vars in 538 bytes} [Thu May 15 10:48:25 2025] POST /api/collections => generated 21 bytes in 1 msecs (HTTP/1.1 500) 1 headers in 64 bytes (1 switches on core 0)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
goodbye to uWSGI.
